src/types/custom.ts

```diff
@@ -0,0 +1,20 @@
+import { Runtype, Rt, Static, create, validationError } from '../runtype'
+import { String } from './string'
+
+export interface Custom<A extends Rt, T extends string, K> extends Runtype<Static<A>> {
+  tag: T,
+  underlying: A,
+  args: K
+}
+
+export function Custom<A extends Rt, T extends string, K>(underlying: A, constraint: (x: Static<A>, args: K) => boolean | string, tag: T, args: K) {
```

Comment:
I thought it was more DRY. See example in test: https://github.com/typeetfunc/runtypes/blob/feature/custom-type/src/index.spec.ts#L60